solr/core/src/java/org/apache/solr/core/SolrCore.java

```diff
@@ -414,7 +416,19 @@ private IndexDeletionPolicyWrapper initDeletionPolicy(IndexDeletionPolicyWrapper
     } else {
       delPolicy = new SolrDeletionPolicy();
     }
-    return new IndexDeletionPolicyWrapper(delPolicy);
+
+    return new IndexDeletionPolicyWrapper(delPolicy, snapshotMgr);
+  }
+
+  private SolrSnapshotMetaDataManager initSnapshotMetaDataManager() {
```

Comment:
What do you think about possibly creating a Directory lazily on first use of a snapshot?  I'm a tad concerned we're weighing Solr down a bit more for everyone when only a small number of installations in the future will use this.  What's your feeling @markrmiller ?